src/app/api/comment/comment.controller.ts

```diff
@@ -0,0 +1,39 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '../core/utils/authenticate'
+import { CommentService } from './comment.service'
+import { CreateCommentSchema, UpdateCommentSchema } from '@/types/dto/comment.dto'
+import httpStatus from 'http-status'
+import { IdParams } from '../core/types/api'
+
+export const createComment = async (req: NextRequest) => {
+  const user = await authenticate(req)
+
+  const commentService = new CommentService(user)
+
+  const data = CreateCommentSchema.parse(await req.json())
+
+  const comment = await commentService.createComment(data)
```

Comment:
since its `commentService` just `create` method name is fine.

Reply:
Updated in 8315c817d2e398c277c36edbadab25a14b0e42fd commit